codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-aws-query-compatible-json-async-client-class.java

```diff
@@ -0,0 +1,177 @@
+                .create("ApiCall");
+        try {
+            apiCallMetricCollector.reportMetric(CoreMetric.SERVICE_ID, "QueryToJsonCompatibleService");
+            apiCallMetricCollector.reportMetric(CoreMetric.OPERATION_NAME, "APostOperation");
+            JsonOperationMetadata operationMetadata = JsonOperationMetadata.builder().hasStreamingSuccessResponse(false)
+                    .isPayloadJson(true).build();
+
+            HttpResponseHandler<APostOperationResponse> responseHandler = protocolFactory.createResponseHandler(
+                operationMetadata, APostOperationResponse::builder);
+
+            HttpResponseHandler<AwsServiceException> errorResponseHandler = createErrorResponseHandler(protocolFactory,
+                                                                                                       operationMetadata);
+            String hostPrefix = "{StringMember}-foo.";
+            HostnameValidator.validateHostnameCompliant(aPostOperationRequest.stringMember(), "StringMember",
+                                                        "aPostOperationRequest");
+            String resolvedHostExpression = String.format("%s-foo.", aPostOperationRequest.stringMember());
+
+            CompletableFuture<APostOperationResponse> executeFuture = clientHandler
+                    .execute(new ClientExecutionParams<APostOperationRequest, APostOperationResponse>()
+                             .withOperationName("APostOperation")
+                             .withMarshaller(new APostOperationRequestMarshaller(protocolFactory))
+                             .withResponseHandler(responseHandler).withErrorResponseHandler(errorResponseHandler)
+                             .withMetricCollector(apiCallMetricCollector).hostPrefixExpression(resolvedHostExpression)
+                             .withInput(aPostOperationRequest));
+            CompletableFuture<APostOperationResponse> whenCompleted = executeFuture.whenComplete((r, e) -> {
+                metricPublishers.forEach(p -> p.publish(apiCallMetricCollector.collect()));
+            });
+            executeFuture = CompletableFutureUtils.forwardExceptionTo(whenCompleted, executeFuture);
+            return executeFuture;
+        } catch (Throwable t) {
+            metricPublishers.forEach(p -> p.publish(apiCallMetricCollector.collect()));
+            return CompletableFutureUtils.failedFuture(t);
+        }
+    }
+
+    @Override
+    public void close() {
+        clientHandler.close();
+    }
+
+    private <T extends BaseAwsJsonProtocolFactory.Builder<T>> T init(T builder) {
+        return builder
+            .clientConfiguration(clientConfiguration)
+            .defaultServiceExceptionSupplier(QueryToJsonCompatibleException::builder)
+            .protocol(AwsJsonProtocol.AWS_JSON)
+            .protocolVersion("1.1")
+            .hasAwsQueryCompatible(true)
```

Comment:
Is `.hasAwsQueryCompatible(true)` the only code being added? Could you show me a diff between client w/o hasAwsQueryCompatible and client w/ hasAwsQueryCompatible?

Reply:
Yes, `.hasAwsQueryCompatible(true)` is the only code being added. This `test-aws-query-compatible-json-async-client-class.java` file was auto generated based on service-2.json file added here: https://github.com/ziyanli-amazon/aws-sdk-java-v2/blob/master/test/codegen-generated-classes-test/src/main/resources/codegen-resources/awsquerycompatible/service-2.json and this service-2.json is a copy and paste of https://github.com/ziyanli-amazon/aws-sdk-java-v2/blob/master/test/codegen-generated-classes-test/src/main/resources/codegen-resources/bearerauth/service-2.json with `awsQueryCompatible` field added.